examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts

```diff
@@ -13,18 +14,11 @@ export class UpdateBpiSubjectCommandHandler
   ) {}
 
   async execute(command: UpdateBpiSubjectCommand) {
-    const bpiSubjectToUpdate =
+    const bpiSubjectToUpdate: BpiSubject =
       await this.agent.fetchUpdateCandidateAndThrowIfUpdateValidationFails(
         command.id,
       );
-
-    this.agent.updateBpiSubject(
-      bpiSubjectToUpdate,
-      command.name,
-      command.desc,
-      command.publicKey,
-    );
-
-    await this.storageAgent.updateBpiSubject(bpiSubjectToUpdate);
+    Object.assign(bpiSubjectToUpdate, command);
```

Comment:
Why is agent method removed and object assign used instead? The goal is to avoid HOW on the level on the command and instead just explain WHAT through declarative agent method. It is the job of the agent method to deal with the actual implemenentation (which is simple object assign for now but will be something more complicated as we move forward.)

Reply:
this method is calling update name, description, publicKey methods on BpiSubject which are doing essentially the same thing as Object.assign

Reply:
The idea is that the state of a domain object is always changed through it's methods. They are simple now, but will become much more complex as we move forward. Also, we introduced the agent approach to make sure that the abstraction level is always the same on the level of the command. You just read the names of the methods and you understand what is happening. HOW is hidden away inside of the agent method, which can further change the state of the domain object through the  exposed methods.

Reply:
as agreed, code changes have been pushed